GeneralsMD/Code/Libraries/Source/WWVegas/WWMath/matrix3d.cpp

```diff
@@ -520,7 +522,9 @@ void Matrix3D::Get_Inverse(Matrix3D & inv) const
 	Matrix4x4	mat4Inv;
 
 	float det;
+#ifdef _WIN32
 	D3DXMatrixInverse((D3DXMATRIX *)&mat4Inv, &det, (D3DXMATRIX*)&mat4);
+#endif
```

Comment:
I think some alternative implementation should be provided or some kind runtime error should be produced...

Reply:
Not sure if this specific line is that, but what happens in the code at many places is a row major to column major matrix conversion or vice versa. We polished this in Thyme and it needs to be done in this code as well. This needs to be done in a separate change because it is bigger change. Here is the reference:

https://github.com/TheAssemblyArmada/Thyme/commit/2114738dca9a9a2f44d866bbda0507a470bc5941

Comment from that change:
```
// When converting Matrix4 to D3DMATRIX or vice versa always use conversion function below.
 
// Reason being, D3DMATRIX is row major matrix, and Matrix4 is column major matrix.
 
// Thus copying from one to another will always require a transpose (or invert).
```

I have opened task with #552